src/map/script.c

```diff
@@ -27941,7 +27941,7 @@ static BUILDIN(openlapineddukddakboxui)
 	struct map_session_data *sd = script_rid2sd(st);
 	if (sd == NULL)
 		return false;
-	const int item_id = script_getnum(st, 2);
+	const int item_id = (script_getnum(st, 2) == 0) ? sd->itemid : script_getnum(st, 2);
```

Comment:
missing parameters need check with ``script_hasdata``
and why it do two times check? may be better read item id once?
also what if ``sd->itemid`` is ``-1``?

Reply:
What do you think if checking item like this?
```c
	struct item_data *it;
	if ( script_hasdata( st, 2 ) ) {
		if ( script_isstring( st, 2 ) ) {
			const char* item_name = script_getstr( st, 2 );
			it = itemdb->search_name(item_name);
		}else{
			it = itemdb->exists(script_getnum( st, 2 ));
		}
	} else {
		// Call function without param
		it = itemdb->exists(sd->itemid);
	}
```

Reply:
almost good
style with ``if`` look like from rathena style. need write like this ``if (script_hasdata(st, 2)) {``

``sd->itemid`` almost always equal to ``-1`` except some cases. this item never exists. may be better if equal ``-1`` return error?

Reply:
@4144 your mean is check `sd->itemid` before check exists, like this?
```c
// Call function without param
if (sd->itemid == -1) {
	return false;
}
it = itemdb->exists(sd->itemid);
```
Because item id will be great then 0, can we check `sd->itemid <= 0` before check item exists?

Reply:
yes